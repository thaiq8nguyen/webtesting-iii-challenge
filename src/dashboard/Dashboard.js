import React from "react";

import Display from "../display/Display";
import Controls from "../controls/Controls";
import UserProfile from "../UserProfile/UserProfile";
import { Grid } from "semantic-ui-react";

class Dashboard extends React.Component {
  state = {
    locked: false,
    closed: false
  };

  render() {
    const { closed, locked } = this.state;

    return (
      <>
        <Grid columns={2} container padded="vertically">
          <Grid.Column textAlign="center">
            <Display locked={locked} closed={closed} />
            <Controls
              locked={locked}
              closed={closed}
              toggleLocked={this.toggleLocked}
              toggleClosed={this.toggleClosed}
            />
          </Grid.Column>
          <Grid.Column>
            <UserProfile />
          </Grid.Column>
        </Grid>
      </>
    );
  }

  toggleLocked = () => {
    this.setState(prev => ({ locked: !prev.locked }));
  };

  toggleClosed = () => {
    this.setState(prev => ({ closed: !prev.closed }));
  };
}

export default Dashboard;
