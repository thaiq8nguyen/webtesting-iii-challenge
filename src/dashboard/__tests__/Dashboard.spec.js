import React from "react";
import { cleanup, render } from "@testing-library/react";
import Dashboard from "../Dashboard";



afterEach(cleanup);

/* SNAPSHOT TESTING */
test("creating controller state snapshot using snapshot testing", () => {
    const { asFragment } = render(<Dashboard/>);
    expect(asFragment()).toMatchSnapshot();
})