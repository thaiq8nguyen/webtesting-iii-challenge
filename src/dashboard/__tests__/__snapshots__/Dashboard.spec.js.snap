// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating controller state snapshot using snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="ui container vertically padded two column grid"
  >
    <div
      class="center aligned column"
    >
      <div
        class="display panel"
      >
        <div
          class="led green-led"
        >
          Unlocked
        </div>
        <div
          class="led green-led"
        >
          Open
        </div>
      </div>
      <div
        class="controls panel"
      >
        <button
          class="toggle-btn"
          data-testid="toggle-lock-gate"
          disabled=""
        >
          Lock Gate
        </button>
        <button
          class="toggle-btn"
          data-testid="toggle-close-gate"
        >
          Close Gate
        </button>
      </div>
    </div>
    <div
      class="column"
    >
      <div>
        <div
          class="ui card"
        >
          <div
            class="content"
          >
            <div
              class="ui two column grid"
            >
              <div
                class="middle aligned row"
              >
                <div
                  class="column"
                >
                  <div
                    class="header"
                  >
                    UserProfile
                  </div>
                </div>
                <div
                  class="right aligned column"
                >
                  <button
                    class="ui button"
                    small="tiny"
                  >
                    Get Profile
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="content"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
